relations/src/hasher.rs

```diff
@@ -0,0 +1,83 @@
+use halo2_base::{
+    gates::GateChip, poseidon::hasher::PoseidonHasher, utils::BigPrimeField, AssignedValue, Context,
+};
+
+use crate::{account::CircuitAccount, note::CircuitNote};
+
+pub trait InnerHasher<F: BigPrimeField> {
```

Comment:
`inner` because it happens in circuit? if so, then I would stick to other modules, where you use `circuit` word

Reply:
Same as above